instrumentation/opentelemetry-instrumentation-django/setup.cfg

```diff
@@ -45,6 +45,8 @@ install_requires =
     opentelemetry-semantic-conventions == 0.23.dev0
 
 [options.extras_require]
+asgi =
+    opentelemetry-instrumentation-asgi == 0.23.dev0
```

Comment:
This should be documented so people using ASGI with Django realize they need to install these extra deps to make it work. 

How do ASGI deps work with Django? Do regular Django projects need to install additional packages to make ASGI work or does Django now ship with (or depend on) bundled ASGI module.  If some level of ASGI support is always present in Django now, may be we can add this as a main dependency instead?

Reply:
This is a good discussion topic. Django doesn't require any additional/extra dependencies for ASGI to work (aside from running it using Uvicorn/Daphne), and bundles `asgiref` since the async introduction in Django 3.0: https://github.com/django/django/commit/a415ce70bef6d91036b00dd2c8544aed7aeeaaed#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7

This dependency doesn't include other heavy dependencies that aren't already required by `opentelemetry-instrumentation-wsgi`. However, we also have to consider that ASGI is only present since Django 3.x, while this project tries to maintain support for older Django versions.

I think the only missing change on this PR, if we go with adding this to `install_requires`, is to make `_is_asgi_supported` also depend on `DJANGO_3_0` being `True`, as we won't receive any `ImportError` anymore when importing ASGI dependencies.